custom_components/waste_collection_schedule/waste_collection_schedule/source/ecoapp_ecoservice_lt.py

```diff
@@ -0,0 +1,174 @@
+            self._waste_object_ids = self._waste_object_ids.split(",")
+        
+        entries = []
+        
+        for obj_id in self._waste_object_ids:
+            if not isinstance(obj_id, str):
+                raise ValueError(
+                    "Waste object IDs must be strings, got: {}".format(type(obj_id))
+                )
+
+            try:
+                # cleanup user string
+                obj_id = obj_id.strip()
+                obj_id = obj_id.replace('"', '')
+                obj_id = obj_id.replace("'", "")        
+                
+                # fetch container data
+                c_query = {
+                    "r": "api/calendar",
+                    "key": "fd1d20bcd36e86504ac4bad2d88a2caa",  # magic API key, not clear when and if it changes
+                    "containerId": obj_id,
+                    "getSpecificStuff": "container_id"
+                }       
+        
+                r = requests.get(
+                    self.API_URL,
+                    params=c_query,
+                )
+                r.raise_for_status()
+                
+                container = json.loads(r.text)
+
+                self.check_for_container_error(container)
+
+                # fetch container collection schedule
+                s_query = {
+                    "r": "api/calendar",
+                    "key": "fd1d20bcd36e86504ac4bad2d88a2caa",  # magic API key, not clear when and if it changes
+                    "containerId": container[0]["key"],
+                    "getSpecificStuff": "container_map"
+                }
+
+                r = requests.get(
+                    self.API_URL,
+                    params=s_query,
+                )
+                r.raise_for_status()
+                
+                schedule = json.loads(r.text)
+
+                if len(schedule) == 0:
+                    raise Exception(
+                        "Error: failed to fetch collection schedule data"
+                    )
+                
+                for date, value in schedule.items():
+                    if value:
+                        entries.append(
+                            Collection(
+                                date=datetime.strptime(date, "%Y-%m-%d").date(),
+                                t=TYPE_MAP.get(container[0]["Type"], {}).get("name"),
+                                icon=TYPE_MAP.get(container[0]["Type"], {}).get("icon"),
+                            )
+                        )
+                    
+                if len(entries) == 0:
+                    raise Exception(
+                        "Error: no collection schedule data found for container ID: {}".format(obj_id)
+                    )  
+            
+            except Exception as e:
+                raise Exception (
+                    f"Error fetching data for waste object ID {obj_id}: {e}"                
+                )
```

Comment:
Best practice to put a from e to the raise so the error is easier to debug.
```suggestion
                raise Exception (
                    f"Error fetching data for waste object ID {obj_id}: {e}"                
                ) from e
```